.github/workflows/test.yml

```diff
@@ -1,30 +1,66 @@
 name: test
-
 on:
   push:
-    branches:
-      - master
+    branches: [master]
+    tags: ['**']
   pull_request:
   workflow_dispatch:
-
-concurrency: 
-  group: ${{ github.head_ref || github.run_id }}
+concurrency:
+  group: ${{ github.ref }}
```

Comment:
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Reply:
I don't quite understand this suggestion.

Currently whenever a branch/tag/PR/ is updated, any previous jobs on the same branch/tag/PR are cancelled.

Do you want something else? Note that[^conc][^gh]:

- there are no other workflows so `github.workflow` isn't needed
- `github.head_ref` is only defined on PRs
- `github.run_id` would have no effect (it's the default behaviour)

[^conc]: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#concurrency
[^gh]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#github-context